### ISSUE TITLE
Pattern is reported twice on pattern/@documents usage

### ISSUE THREAD

**Issue by xatapult** [schxslt/schxslt#210]
When I use a `pattern/@documents` construction, the pattern that contains the `@documents` seems to fire twice...

To reproduce, from the zip: Validate `subdocuments-master.xml` with `subdocuments-1.sch`. I get the following SVRL with the pattern seeming to fire twice where only once is expected (The ISO implementation only returns a single one).

```xml
<svrl:schematron-output xmlns:svrl="http://purl.oclc.org/dsdl/svrl"
  xmlns:sch="http://purl.oclc.org/dsdl/schematron"
  xmlns:error="https://doi.org/10.5281/zenodo.1495494#error"
  xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
  xmlns:schxslt-api="https://doi.org/10.5281/zenodo.1495494#api"
  xmlns:schxslt="https://doi.org/10.5281/zenodo.1495494" xmlns:xs="http://www.w3.org/2001/XMLSchema">
  <svrl:metadata xmlns:dct="http://purl.org/dc/terms/"
    xmlns:skos="http://www.w3.org/2004/02/skos/core#">
    <dct:creator>
      <dct:Agent>
        <skos:prefLabel>SAXON/EE 9.9.1.7</skos:prefLabel>
      </dct:Agent>
    </dct:creator>
    <dct:created>2021-09-06T10:30:51.052+02:00</dct:created>
    <dct:source>
      <rdf:Description xmlns:dc="http://purl.org/dc/elements/1.1/">
        <dct:creator>
          <dct:Agent>
            <skos:prefLabel>SchXslt/1.8 SAXON/EE 9.9.1.7</skos:prefLabel>
            <schxslt.compile.typed-variables xmlns="https://doi.org/10.5281/zenodo.1495494#"
              >true</schxslt.compile.typed-variables>
          </dct:Agent>
        </dct:creator>
        <dct:created>2021-09-06T10:30:51.034+02:00</dct:created>
      </rdf:Description>
    </dct:source>
  </svrl:metadata>
  <svrl:active-pattern documents="file:/C:/xdata2/test-pattern-documents/subdocuments-1.xml"/>
  <svrl:fired-rule context="p"/>
  <svrl:fired-rule context="p"/>
  <svrl:failed-assert location="/Q{}document[1]/Q{}p[1]" test="@class ne 'donotuse'">
    <svrl:text> The class donotuse is forbidden
      (file:/C:/xdata2/test-pattern-documents/subdocuments-2.xml)! </svrl:text>
  </svrl:failed-assert>
  <svrl:active-pattern documents="file:/C:/xdata2/test-pattern-documents/subdocuments-2.xml"/>
  <svrl:fired-rule context="p"/>
  <svrl:fired-rule context="p"/>
  <svrl:failed-assert location="/Q{}document[1]/Q{}p[1]" test="@class ne 'donotuse'">
    <svrl:text> The class donotuse is forbidden
      (file:/C:/xdata2/test-pattern-documents/subdocuments-2.xml)! </svrl:text>
  </svrl:failed-assert>
</svrl:schematron-output>
```

**Comment by xatapult** [comment]
[test-pattern-documents.zip](https://github.com/schxslt/schxslt/files/7114491/test-pattern-documents.zip)

**Comment by dmj** [comment]
I can confirm that there is a bug. The pattern is not applied/activated but reported twice. 

I added a xsl:message to the xsl:if condition in the compiled validation stylesheet and it reports triggers only once.

**Comment by dmj** [comment]
Thanks for reporting this issue! It was fixed in 1.8.1 released just now.
